lib/pf/access_filter/radius.pm

```diff
@@ -0,0 +1,166 @@
+package pf::access_filter::radius;
+
+=head1 NAME
+
+pf::access_filter::radius - handle the answer of the radius request
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::access_filter::radius deny, rewrite radius answer based on rules.
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+use pf::api::jsonrpcclient;
+use pf::violation qw (violation_view_top);
+use pf::locationlog qw(locationlog_set_session);
+
+use base qw(pf::access_filter);
+tie our %ConfigRadiusFilters, 'pfconfig::cached_hash', 'config::RadiusFilters';
+tie our %RadiusFilterEngineScopes, 'pfconfig::cached_hash', 'FilterEngine::RadiusScopes';
+
+
+=head1 SUBROUTINES
+
+=over
+
+=item new
+
+=cut
+
+sub new {
+   my $logger = Log::Log4perl::get_logger("pf::access_filter::radius");
+   $logger->debug("instantiating new pf::access_filter::radius");
+   my ( $class, %argv ) = @_;
+   my $self = bless {}, $class;
+   return $self;
+}
+
+=item test
+
+Test all the rules
+
+=cut
+
+sub test {
+    my ($self, $scope, $args) = @_;
+    if (exists $RadiusFilterEngineScopes{$scope}) {
+       $args->{'violation'} = violation_view_top($args->{'mac'});
+       return $RadiusFilterEngineScopes{$scope}->match_first($args);
+    }
+    return undef;
+}
+
+=head2 handleRoleInRule
+
+Handles the role update
+
+=cut
+
+sub handleAnswerInRule {
+    my ($self, $rule, $args) = @_;
+    my $logger = Log::Log4perl::get_logger( ref($self) );
+    my $radius_reply_ref = {};
+    if(defined $rule) {
+        $radius_reply_ref = {'Reply-Message' => "Inverse is the best"};
```

Comment:
I think the one you chose is a bit more professional
Plus, if it's not one of Zaymit servers then the statement is not true which will surely create other bugs (pf magic)

Lets leave it as is then